lib/ratonvirus/storage/carrierwave.rb

```diff
@@ -20,7 +20,18 @@ def asset_path(asset)
         return if asset.nil?
         return if asset.file.nil?
 
-        yield asset.file.path
+        if asset.file.is_a?(CarrierWave::Storage::Fog::File)
+          # We can't use carrierwave_uploader.cache_stored_file! as this is
+          # remote too when using fog/s3.
+          @tempfile = Tempfile.new(encoding: "ascii-8bit").tap do |f|
+            f.write(asset.read)
+            f.close
+          end
+
+          yield @tempfile.path
```

Comment:
This would leave the tempfile stored in the file system after the check.

Please see an example how this is implemented for active storage:
https://github.com/mainio/ratonvirus/blob/d6d0313dc872e8fc859b1a4a7af5a15d090a9eec/lib/ratonvirus/storage/active_storage.rb#L117-L126

Could you please change this to work similarly?

After you've yielded the filepath, you can delete it. The underlying block that receives the tempfile's path will run the can after which the tempfile can be removed.

Reply:
Also, I don't think we need to store the tempfile in an instance variable. It's not being used/accessed later.